publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -305,9 +308,29 @@ public Resource updateResource(Resource resource, UserInstance userInstance) {
 
     // update this method according to current needs.
     public Entity migrate(Entity dataEntry) {
+        if (dataEntry instanceof Resource resource) {
+            persistPublicationChannelForPublisherWhenDegree(resource);
+        }
         return dataEntry;
     }
 
+    private void persistPublicationChannelForPublisherWhenDegree(Resource resource) {
+        resource.getPublisherWhenDegree().ifPresent(publisher -> {
```

Comment:
Do you write the Resource to mark it as "refreshed"?

Reply:
Version will be updated when Resource will be refreshed/ persisted back in `EventBasedBatchScanHandler`.